src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1202,12 +1202,11 @@ ucs_status_t uct_dc_mlx5_iface_fc_handler(uct_rc_iface_t *rc_iface, unsigned qp_
         status = uct_dc_mlx5_iface_fc_grant(&dc_req->super.super);
         if (status == UCS_ERR_NO_RESOURCE){
             uct_dc_mlx5_ep_do_pending_fc(ep, dc_req);
-            status = UCS_OK;
         } else if (status != UCS_OK) {
             ucs_diag("fc_ep %p: failed to send %s: %s", ep,
-                     uct_dc_mlx5_iface_fc_pure_grant_str(
-                             dc_req, buf, ucs_static_array_size(buf)),
+                     uct_dc_mlx5_fc_req_str(dc_req, buf, sizeof(buf)),
                      ucs_status_string(status));
+            return status;
```

Comment:
when this function returns error instead of OK, seems it's causing this:
https://github.com/openucx/ucx/blob/ab13538811ffe55eab4cf7058c50ded1fe1c596b/src/uct/ib/rc/accel/rc_mlx5.inl#L120
is this really what needs to happen?
maybe make fc_handler return void?

Reply:
nice catch, didn't notice it.
shall we fix RC and DC handlers to always return UCS_OK, except the case when RC handles AM inside it

Reply:
yes

Reply:
and maybe add comment inside the handlers to not miss it

Reply:
> and maybe add comment inside the handlers to not miss it

yes, makes sense it